immer/detail/hamts/champ_iterator.hpp

```diff
@@ -10,11 +10,38 @@
 
 #include <immer/detail/hamts/champ.hpp>
 #include <immer/detail/iterator_facade.hpp>
+#include <immer/relocation/gc_relocation_policy.hpp>
 
 namespace immer {
 namespace detail {
 namespace hamts {
 
+template <typename Policy, bits_t B>
+struct relocation_info_t
+{
+    void reset() {}
+    void increment() {}
+    void step_right(count_t) {}
+    void step_down(count_t) {}
+};
+
+template <bits_t B>
+struct relocation_info_t<gc_relocation_policy, B>
+{
+    count_t cur_off_;
+    std::uint8_t path_off_[max_depth<B>] = {
+        0,
+    };
+
+    void reset() { cur_off_ = 0; }
+
+    void increment() { cur_off_++; }
+
+    void step_right(count_t depth) { ++path_off_[depth - 1]; }
+
+    void step_down(count_t depth) { path_off_[depth - 1] = 0; }
+};
```

Comment:
Nice, I like how you have managed here to extract out into a policy the bits needed and avoid duplication!